x-pack/plugins/canvas/public/expression_types/datasource.tsx

```diff
@@ -62,10 +62,18 @@ const DatasourceWrapper: React.FunctionComponent<DatasourceWrapperProps> = (prop
 
   useEffect(() => {
     callRenderFn();
-    return () => {
+  }, [callRenderFn, props]);
+
+  useEffect(
+    () => () => {
       handlers.destroy();
-    };
-  }, [callRenderFn, handlers, props]);
+    },
+    // newly created handlers are coming from the component, which is rendering
+    // and returned function from 'useEffect' is calling for cleanup, not on unmount,
+    // but 'destroy' need to be called on unmount of the component
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
```

Comment:
Should this include `handlers.destroy` as a dependency? 

In the commit before we moved to hooks, it was `componentWillUnmount => props.handlers.destroy` so should it always be using the "newest" `handlers.destroy`?